cli/tests/unit/test_match_hashes.py

```diff
@@ -0,0 +1,163 @@
+    assert match_1.pattern_hash == match_2.pattern_hash
+
+
+@pytest.mark.quick
+def test_code_hash_independent_of_rulename(mocker, foo_contents):
+    mocker.patch.object(Path, "open", mocker.mock_open(read_data=foo_contents))
+    match_1 = get_rule_match(rule_id="first.rule.id")
+    match_2 = get_rule_match(rule_id="second.rule.id")
+    assert match_1.syntactic_id != match_2.syntactic_id
+    assert match_1.match_based_id != match_2.match_based_id
+    assert match_1.code_hash == match_2.code_hash
+    assert match_1.pattern_hash == match_2.pattern_hash
+
+
+@pytest.mark.quick
+def test_code_hash_independent_of_index(mocker, eqeq_rule, foo_contents):
+    mocker.patch.object(Path, "open", mocker.mock_open(read_data=foo_contents))
+    match_1 = get_rule_match(start_line=3, end_line=3)
+    match_2 = get_rule_match(start_line=4, end_line=4)
+    matches = RuleMatchSet(eqeq_rule)
+    matches.update([match_1, match_2])
+    matches = list(sorted(matches))
+    # Adding a RuleMatch to a RuleMatchSet does not update the index of the
+    # original RuleMatch object, instead it creates a copy of each RuleMatch
+    # that gets added to the set, and updates those indexes
+    assert matches[0].index == 0
+    assert matches[1].index == 1
+    assert matches[0].syntactic_id != matches[1].syntactic_id
+    assert matches[0].match_based_id != matches[1].match_based_id
+    assert matches[0].code_hash == matches[1].code_hash
+    assert matches[0].pattern_hash == matches[1].pattern_hash
+
+
+@pytest.mark.quick
+def test_code_hash_changes_with_code(mocker, eqeq_rule, foo_contents):
+    mocker.patch.object(Path, "open", mocker.mock_open(read_data=foo_contents))
+    match_1 = get_rule_match(
+        start_line=3, end_line=3, metavars={"$X": {"abstract_content": "5"}}
+    )
+    match_2 = get_rule_match(
+        start_line=5, end_line=5, metavars={"$X": {"abstract_content": "6"}}
+    )
+    matches = RuleMatchSet(eqeq_rule)
+    matches.update([match_1, match_2])
+    matches = list(sorted(matches))
+    # Adding a RuleMatch to a RuleMatchSet does not update the index of the
+    # original RuleMatch object, instead it creates a copy of each RuleMatch
+    # that gets added to the set, and updates those indexes
+    assert matches[0].index == 0
+    assert matches[1].index == 0
+    assert matches[0].code_hash != matches[1].code_hash
+    assert matches[0].pattern_hash != matches[1].pattern_hash
+
+
+@pytest.mark.quick
+def test_line_hashes_hash_correct_line(mocker, double_eqeq_rule, foo_contents):
+    mocker.patch.object(Path, "open", mocker.mock_open(read_data=foo_contents))
+    match_1 = get_rule_match(start_line=4, end_line=5)  # 5 == 5\n6 == 6\n
+    match_2 = get_rule_match(start_line=5, end_line=6)  # 6 == 6\n5 == 5\n
+    matches = RuleMatchSet(double_eqeq_rule)
+    matches.update([match_1, match_2])
+    matches = list(sorted(matches))
+    assert matches[0].start_line_hash != matches[0].end_line_hash
```

Comment:
Thanks for adding these checks since it doesn't look like snapshots have coverage of these.